test/val/val_id_test.cpp

```diff
@@ -941,7 +941,299 @@ TEST_F(ValidateID, OpSpecConstantBad) {
   CHECK(spirv, SPV_ERROR_INVALID_BINARY);
 }
 
-// TODO: OpSpecConstantComposite
+// Valid: SpecConstantComposite specializes to a vector.
+TEST_F(ValidateID, OpSpecConstantCompositeVectorGood) {
+  const char* spirv = R"(
+%1 = OpTypeFloat 32
+%2 = OpTypeVector %1 4
+%3 = OpSpecConstant %1 3.14
+%4 = OpConstant %1 3.14
+%5 = OpSpecConstantComposite %2 %3 %3 %4 %4)";
+  CHECK(spirv, SPV_SUCCESS);
+}
+
+// Valid: Vector of floats and Undefs.
+TEST_F(ValidateID, OpSpecConstantCompositeVectorWithUndefGood) {
+  const char* spirv = R"(
+%1 = OpTypeFloat 32
+%2 = OpTypeVector %1 4
+%3 = OpSpecConstant %1 3.14
+%5 = OpConstant %1 3.14
+%9 = OpUndef %1
+%4 = OpSpecConstantComposite %2 %3 %5 %3 %9)";
+  CHECK(spirv, SPV_SUCCESS);
+}
+
+// Invalid: result type is float.
+TEST_F(ValidateID, OpSpecConstantCompositeVectorResultTypeBad) {
+  const char* spirv = R"(
+%1 = OpTypeFloat 32
+%2 = OpTypeVector %1 4
+%3 = OpSpecConstant %1 3.14
+%4 = OpSpecConstantComposite %1 %3 %3 %3 %3)";
+  CHECK(spirv, SPV_ERROR_INVALID_ID);
```

Comment:
Let's check message substrings as well.  That way it's easier to tell if we've covered all the error cases in the code.